department-of-veterans-affairs/abd-vro
# MCP-1986 audit event persistence model

## What was the problem?

We need a way to track the history of a manual or automated claim. 

Associated tickets or Slack threads:

- [MCP-1986](https://amida.atlassian.net/browse/MCP-1986)

## How does this fix it?[^1]

This PR introduces the following enhancements to the audit event framework:

- audit events are persisted in the database
- The framework allows extracting information from the Auditable object and storing this information in a "details" column
- Mas endpoints are recording audit events

## How to test this PR
- Call automatedClaim endpoint from Swagger
- Examine the database: (select * from audit_event) to see the history of this claim
- Do the same for examOrderingStatus

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
